realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/UserTests.kt

```diff
@@ -183,34 +179,34 @@ class UserTests {
         assertTrue(anonUser === linkedUser)
         assertEquals(2, linkedUser.identities.size)
         assertEquals(Credentials.IdentityProvider.EMAIL_PASSWORD, linkedUser.identities[1].provider)
-        admin.setAutomaticConfirmation(enabled = true)
 
         val otherEmail = TestHelper.getRandomEmail()
         val otherPassword = "123456"
         app.emailPasswordAuth.registerUser(otherEmail, otherPassword)
-        linkedUser = anonUser.linkCredentials(Credentials.emailPassword(email, password))
-        assertTrue(anonUser === linkedUser)
-        assertEquals(3, linkedUser.identities.size)
-        assertEquals(Credentials.IdentityProvider.EMAIL_PASSWORD, linkedUser.identities[2].provider)
```

Comment:
Users are limited to 2 credentials and this test was testing with three.

Modified it to validate that it should fail with three.

Reply:
I'm not sure whether this is expected or not. Maybe ask someone from the server team about adding more than one userpass credential and about the existence of limitations in the types/amount of credentials being linked?

Reply:
Have included tests for **serverKey**, **apiKey** and **customFunction**. 

**serverKey** and **apiKey** fail to run yielding the following errors:

- serverKey error: `AUTH_ERROR(realm::app::ServiceError:47): invalid API key`
- apiKey error: `INVALID_PARAMETER(realm::app::ServiceError:6): invalid user link request`